crates/sparrow-instructions/src/evaluators/math.rs

```diff
@@ -179,12 +152,16 @@ where
     T: NativeFromScalar,
 {
     fn evaluate(&mut self, info: &dyn RuntimeInfo) -> anyhow::Result<ArrayRef> {
-        let lhs = info.value(&self.lhs)?.primitive_array()?;
+        let lhs = info.value(&self.lhs)?.array_ref()?;
         let rhs = info.value(&self.rhs)?;
         let result = match rhs.try_primitive_literal::<T>() {
             Ok(Some(rhs)) if !rhs.is_zero() => {
                 // Division by a literal
-                Arc::new(arrow::compute::divide_scalar(lhs.as_ref(), rhs)?)
+                let rhs: arrow_array::PrimitiveArray<T> =
+                    arrow_array::PrimitiveArray::from_value(rhs, 1);
+                let rhs = arrow_array::Scalar::new(&rhs);
+                let result = arrow_arith::numeric::div(&lhs, &rhs)?;
```

Comment:
Could add a helper for creating the static somewhere, seen it in a few instances now

Reply:
Could. I *think* Arrow may be adding one, and I want to take some time. I'm not sure if we just want a helper for creating them, or some kind of serializable wrapper that would allow us to delete our `ScalarValue` class more entirely. I'll file a ticket.

Reply:
Also found `Int64Array::new_scalar(value)` which helps for now. But still filed ticket to track replacing the `ScalarValue` class with `Scalar<ArrayRef>`.